packages/grid/x-grid-modules/src/hooks/features/usePagination.ts

```diff
@@ -149,41 +158,65 @@ export const usePagination = (
   }, [state]);
 
   useEffect(() => {
-    if (rows.length !== state.rowCount) {
+    if (apiRef.current?.isInitialised) {
+      apiRef.current!.emit(PAGE_CHANGED_EVENT, stateRef.current);
+    }
+  }, [apiRef, stateRef, apiRef.current?.isInitialised]);
+
+  useEffect(() => {
```

Comment:
Potentially, a boy scout opportunity (leaving the space in an even better place than we first found it)

```suggestion
  React.useEffect(() => {
```

Reply:
We can do if you want but there is no jsx element in this file. So I don't see any value in doing that.
The code is not cleaner, it's more verbose.

Reply:
I don't think that verbosity is important at this point, it's still short. I think that until the React import is tree-shakable, the standard should be import * from React, so we remove the requirement around changing imports as we refactor the logic (adding named exports, removing them).

I would honestly be willing to have React as a global as even simpler, but I don't think most people are ready for this. Notice how UI libraries works with Vue, default is to put them all as global. I love how you can kill all imports with https://github.com/vuetifyjs/vuetify-loader.

Reply:
> so we remove the requirement around changing imports as we refactor the logic (adding named exports, removing them).

For me this is not an issue at all as I do it for all other libraries. Up to you.